### ISSUE TITLE
Write Tag (Write Block) problems

### ISSUE THREAD

**Issue by portmankim** [ikarus23/MifareClassicTool#75]
This App is cool, I can use it to read some Mifare 1K tags I have here.
I'm using a new Huawei P8 Lite mobile phone, works fine...except:
- I can read all data from Tags, fast and precise
- I can write data to a Tag via Write Dump (Clone), by selecting 1 Sector a time
- but I can't write data using Write Tag -> Write Block:
- \* when using Write Block the App says: "Data Succesfully Written" ..but nothing
- \* when I read back the Tag no data has changed
- \* the Tags are empty (from manufacturer, all keys are FFFF...)
- \* as said before, via Write Dump the data does change on the Tags!!!

What am I doing wrong?
Kim

**Comment by ikarus23** [comment]
Hi Kim,

this is really strange. What Android version is in use? Do you have another NFC enabled phone to check if there is the same issue?

**Comment by psg69** [comment]
Same problem here.
Samsung Galaxy S5...

**Comment by psg69** [comment]
And when it does write via "Write Dump" it is stuffing up and destroying access to all blank sectors...

**Comment by nestornino** [comment]
Hi, I'm currently developing an application to write NFC tags, but have a problem when I want to make writing with keys. Reviewing the code Available Project, found in WriteTag.java something I do not understand. The WriteBlock operation (lines 485 and 492), it has arguments that do not match those of the android developers page, which are an integer and a byte just:
http://developer.android.com/intl/es/reference/android/nfc/tech/MifareClassic.html#

I am new to these issues, but I would help her answer.
Thank you

**Comment by ikarus23** [comment]
Hi, the `writeBlock()` you see is not the function from the [Android API](https://developer.android.com/intl/es/reference/android/nfc/tech/MifareClassic.html#writeBlock%28int,%20byte[]%29). It is the `writeBlock()` of the [MCReader](https://github.com/ikarus23/MifareClassicTool/blob/master/Mifare%20Classic%20Tool/app/src/main/java/de/syss/MifareClassicTool/MCReader.java#L400) (part of MCT). I hope this helps

**Comment by ikarus23** [comment]
Any news on that?

**Comment by nestornino** [comment]
Hello, I discovered that the problem was caused because the documentation for Android developers said that for writing to each block, the numbering of each sector was restarted, and I was trying to write in block manufacturer, so it was the error.

I finished the application, but I put aside, because I want to establish a business model through sale tag in my country.

Thanks for asking about my project.
